### PR TITLE
Upgrade React Select

### LINKED ISSUE #16023: Upgrade react-select
Our react-select library is one major version out of date, and we rely on a `@types` library which isn't wholly accurate. Notably we have a `@ts-expect-error` within our usage of it because the types are overly-specific and incorrect.

We should upgrade to the latest react-select (5.4.0 at the time of writing) which does not require an `@types` library and fix the bugs that occur.

We have certain usages wrapped up in styled components and we should remove the usage of styled-components.

I expect this to be a non-trivial upgrade. [Migration guide](https://react-select.com/upgrade)

### PR DESCRIPTION
**Pull request by krishnaglick**
Resolves #16023

The old react-select setup made a lot of assumptions, abused certain facets of the library, and had some weird typing errors.
This upgrade is intended to put us in a better position for changes and improvements in the future. A number of the assumptions that were made still exist and will likely need a full rewrite in the future to increase maintainability.

For the future of our use of react-select, I feel like we should be able to get away from the whole `components={components}` and instead reference things more simply, and with better types. That would be a larger refactor that what's reasonable here however.

### CODE REVIEW

File: airbyte-webapp/src/components/base/DropDown/SelectContainer.tsx [hunk 1]
```diff
@@ -3,7 +3,14 @@ import { components, ContainerProps } from "react-select";
 
 import { OptionType } from "./DropDown";
 
-export const SelectContainer: React.FC<ContainerProps<OptionType, false>> = (props) => {
+export const SelectContainer: React.FC<
+  ContainerProps<OptionType, false> & {
+    selectProps: {
+      "data-testid": string;
```
Comment (tealjulia) [inline airbyte-webapp/src/components/base/DropDown/SelectContainer.tsx:9]: Using an `aria-label` would be more accessible than a `data-testid` and should still be findable by React Testing Library and Cypress.

**Comment by tealjulia** [comment]
A few notes:
- The issue mentioned nixing the styled components for dropdowns, but I still see `CustomSelect` is a styled component.
- Wow a bunch of the types got more complex... but seemingly more accurate as to what they're actually representing
- Wondering what you believe in here should be a future rewrite.  You mentioned it in the PR and I'm wondering what stands out to you here.

+ left one other specific comment about accessibility/testing

File: airbyte-webapp/src/components/base/DropDown/SelectContainer.tsx (hunk 1 above)
Comment (krishnaglick) [reply airbyte-webapp/src/components/base/DropDown/SelectContainer.tsx:9]: https://github.com/airbytehq/airbyte/blob/6b5e4fb6e6975d833f7c9054fc67e81ec1d53f8f/airbyte-webapp/src/views/Connector/ServiceForm/ServiceForm.test.tsx#L323
We're using this utility that doesn't offer an `aria-label` option. Changing this breaks the tests. Suggestions?

File: airbyte-webapp/src/views/Connection/CatalogTree/components/SyncSettingsDropdown.tsx [hunk 2]
```diff
@@ -24,7 +24,8 @@ const Separator = styled.div`
   padding: 0 5px;
 `;
 
-const SingleValue: React.FC<IProps> = (props) => {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const SingleValue: React.FC<IProps<any>> = (props) => {
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connection/CatalogTree/components/SyncSettingsDropdown.tsx:28]: I think the weird component prop-drilling could be refactored out so these could be correctly typed.

File: airbyte-webapp/src/components/base/DropDown/SelectContainer.tsx (hunk 1 above)
Comment (tealjulia) [reply airbyte-webapp/src/components/base/DropDown/SelectContainer.tsx:9]: We could query by role and use `{name: ${ariaLabelValue}}` probably.  However, this whole form appears to utilize `data-testid` widely so sounds like a bigger project for another time!

**Comment by krishnaglick** [comment]
> A few notes:
> 
>     * The issue mentioned nixing the styled components for dropdowns, but I still see `CustomSelect` is a styled component.
> 
>     * Wondering what you believe in here should be a future rewrite.  You mentioned it in the PR and I'm wondering what stands out to you here.

1. Fixed, though it was less clean than I'd like.
2. I updated the PR description to include my thoughts.

**Review by teallarson — CHANGES_REQUESTED** [review]
Tested locally but haven't looked deeply at the code yet.

Looks like at least this dropdown on connector setup is missing a handful of styles:
Screen Shot 2022-09-08 at 9 04 54 AM

File: airbyte-webapp/src/views/Connection/CatalogTree/components/SyncSettingsDropdown.tsx (hunk 2 above)
Comment (krishnaglick) [reply airbyte-webapp/src/views/Connection/CatalogTree/components/SyncSettingsDropdown.tsx:28]: https://github.com/airbytehq/airbyte/issues/16439

**Comment by krishnaglick** [comment]
> Tested locally but haven't looked deeply at the code yet.
> 
> Looks like at least this dropdown on connector setup is missing a handful of styles: Screen Shot 2022-09-08 at 9 04 54 AM

Pulled out the styled components removal work as it quickly got overly complex. This is "fixed"

File: airbyte-webapp/src/views/Connection/CatalogTree/components/SyncSettingsDropdown.tsx (hunk 2 above)
Comment (timroes) [reply airbyte-webapp/src/views/Connection/CatalogTree/components/SyncSettingsDropdown.tsx:28]: While I agree with splitting this apart if possible, is there any way we can go with `unknown` for now. The `any` here currently actually allows us to access `props.data.value` below, while `props.data` could actually be undefined (at least according to types) and then throw us an exception at runtime.

Comment (krishnaglick) [reply airbyte-webapp/src/views/Connection/CatalogTree/components/SyncSettingsDropdown.tsx]: Changed!

**Review by teallarson — APPROVED** [review]
Tested locally and did not find any visual or functional regression

**Pull request merged**